src/app.cpp

```diff
@@ -101,54 +126,181 @@ int publishLocation(GnssPositioningInfo position) {
+        }
+        publishState = AppPublishState::WaitForConnnect;
+    }
 
-        switch(state) {
-            case AppState::GetGNSSLocation:
+    // Attempt to publish
+    if (millis() - lastPublish > PUBLISH_INTERVAL) {
+        // Handle publish differently based on connection type, also publish location
+        switch (publishState) {
+            case AppPublishState::WaitForConnnect:
             {
-                satellite.getGNSSLocation();
-                // Make sure we re-connect to Skylo NTN after getting gnss fix
-                satellite.process(true /* force updateRegistration */);
-
-                state = AppState::PublishGNSSLocation;
+                if (satellite.connected() || Particle.connected()) {
+                    publishState = AppPublishState::GetGNSSLocation;
+                }
                 break;
             }
 
-            case AppState::PublishGNSSLocation:
+            case AppPublishState::GetGNSSLocation:
             {
-                Log.info("PUBLISH: satellite/1 {\"count\",%d} ------------------", publishCount);
+                satellite.getGNSSLocation();
+                if (modem.radioEnabled() == RADIO_SATELLITE) {
+                    // Make sure we re-connect to Skylo NTN after getting gnss fix
+                    satellite.process(true /* force updateRegistration */);
+                }
+                lastPublish = millis() - PUBLISH_INTERVAL + 2000; // Ensure we don't try to publish immediately after using GNSS
+                publishState = AppPublishState::PublishGNSSLocation;
+                break;
+            }
 
+            case AppPublishState::PublishGNSSLocation:
+            {
                 Variant data;
                 data.set("count", publishCount);
                 data.set("lat", satellite.lastPositionInfo().latitude);
                 data.set("long", satellite.lastPositionInfo().longitude);
                 //data.set("alt", (int)satellite.lastPositionInfo().altitude);
 
-                if (satellite.connected()) {
+                if (satellite.connected())
+                {
+                    Log.info("SATELLITE PUBLISH: {\"count\",%d} ------------------", publishCount);
                     auto satPublishResult = satellite.publish(1 /* code */, data);
                     satPublishResult < 0 ? satPublishFailures++ : satPublishSuccess++;
-                    Log.info("Satellite publish successes/total %d/%d ", satPublishSuccess, publishCount);
+                    Log.info("Satellite publish successes/total %d/%d ", satPublishSuccess, satPublishSuccess + satPublishFailures);
+                    lastPublish = millis();
                 }
+                else if (Particle.connected())
+                {
+                    Log.info("CELLULAR PUBLISH: {\"count\",%d} ------------------", publishCount);
+                    Particle.publish("cellular", String::format("{\"count\",%d}", publishCount));
```

Comment:
Why is the data being published different between satellite and cellular?

Reply:
Its hard to demonstrate but here is what it looks like. 
![ntn_loc_vs_cellular_loc](https://github.com/user-attachments/assets/c344fdbb-dd85-4410-a468-0f7c6a2b3609)

Reply:
I think publishing a variant over the constrained protocol isnt quite the same thing as publishing a cloud event with the `loc` name via the normal `Particle.publish()` interface. Would take some more digging to understand the cloud side of things